src/kernel/caches.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_KERNEL_CACHES_H
+#define BITCOIN_KERNEL_CACHES_H
+
+#include <txdb.h>
+#include <util/check.h>
+
+#include <algorithm>
+#include <bit>
+#include <cstdint>
+#include <limits>
+
+//! Guard against truncation of values before converting.
+constexpr size_t MiBToBytes(int64_t mib)
+{
+    Assert(std::countl_zero(static_cast<uint64_t>(mib)) >= 21); // Ensure sign bit is unset + enough zeros to shift.
+    const int64_t bytes{mib << 20};
+    Assert(static_cast<uint64_t>(bytes) <= std::numeric_limits<size_t>::max());
+    return static_cast<size_t>(bytes);
```

Comment:
In commit "kernel: Move kernel-specific cache size options to kernel" (41bc7164880ffd0782ff5882211c4fcadd656022)

This seems like it is overthinking things. I think this should just be:

```c++
Assert(mib >= 0);
Assert(mib <= std::numeric_limits<size_t>::max() >> 20);
return mib << 20;
```